src/thermo.cpp

```diff
@@ -400,42 +400,56 @@ void Thermo::call_vfunc(int ifield_in)
 
 /* ----------------------------------------------------------------------
    check for lost atoms, return current number of atoms
+   also could number of warnings across MPI ranks and update total
 ------------------------------------------------------------------------- */
 
 bigint Thermo::lost_check()
 {
-  // ntotal = current # of atoms
+  // ntotal = current # of atoms, and Error class warnings
 
-  bigint ntotal;
-  bigint nblocal = atom->nlocal;
-  MPI_Allreduce(&nblocal,&ntotal,1,MPI_LMP_BIGINT,MPI_SUM,world);
-  if (ntotal < 0)
+  bigint nlocal[2], ntotal[2] = {0,0};
+  nlocal[0] = atom->nlocal;
+  nlocal[1] = error->get_numwarn();
+  MPI_Allreduce(nlocal,ntotal,2,MPI_LMP_BIGINT,MPI_SUM,world);
```

Comment:
Does this mean the count of warnings issued is only updated every thermo output step?  If so, that is probably fine, but it is not what the thermo_mod warn doc page says, right?  If so, I think the doc page should be more clear.  Since for thermo 10000 you could get a zillion WARNINGS before the next thermo output, is this a conceptual problem?

Reply:
No. There are two checks. Each MPI rank counts the warnings (numwarn) and it is periodically (when LAMMPS also checks for lost atoms) summed into a total sum (allwarn). The printing of warning messages stops when either counter reaches the limit.

Reply:
Do you mean each proc stops when its local count reaches (e.g.) 100?  In which case (if thermo output is rare) and you are running on 1000 MPI tasks, you will get 100,000 warnings to the screen and logfile?  If so, that's not what the thermo_modify warn doc page would make you think.

Reply:
yes and no. Each MPI rank will stop when either the local count is over the limit **or** the global count. and the global count is updated whenever the lost atom check is done. so, you could indeed construct a case where there are warnings on *all* MPI ranks and no thermo updates where there will be more than the given limit. but that is a rare exception. a more likely  case is that there are either warnings only on a few MPI ranks or regular updates of the global count so the number of warnings happening will be very close to the given limit. and at any rate, you won't be worse off than you currently are with no option at all to control the number. in the pathological scenario, you have now at least the option to (temporarily) turn off warnings altogether.

I wanted to avoid adding an additional and more frequent call to MPI_Allreduce() at all cost.

Reply:
I agree no change to the code is fine, due to the MPI_Allreduce.  But I think the doc page should add a sentence to say what the limit actually means.  Currently it sounds like 100 is the most warnings you will see.

Reply:
@sjplimp I am open to suggestions for improving the docs. But as you can see from the discussion here, it is quite complicated to describe all details very specifically and in most cases it won't make a (big) difference. My thinking was that it will be more effective to not be 100% precise with the description (and then have to explain if somebody has one of those exceptional cases) so that it doesn't confuse people too much that have "normal" situations.